app/task_routes.py

```diff
@@ -0,0 +1,190 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+import requests
+from datetime import datetime
+import os
+
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+
+#---------------------------------------GET------------------------------------------------
+
+def get_one_task_or_abort(task_id):
+    try:
+        task_id = int(task_id)
+    except ValueError:
+        response_str = f"Invalid task ID: {task_id}. ID must be an integer."
+        abort(make_response(jsonify({"Message":response_str}), 400))
+    
+    matching_task = Task.query.get(task_id)
+
+    if not matching_task:
+        response_str = f"Task with id #{task_id} was not found in the database."
+        abort(make_response(jsonify({"Message": response_str}), 404))
+
+    return matching_task
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+    
+    sort_param = request.args.get("sort")
+    if sort_param == "asc":        
+        tasks = Task.query.order_by(Task.title.asc()).all()
+    elif sort_param is None:
+        tasks = Task.query.all()
+    elif sort_param == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+
+    response = []
+    for task in tasks:
+        task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+        response.append(task_dict)
+    return jsonify(response), 200
+    
+    
+    name_param = request.args.get("title")
+
+    if name_param is None:
+        tasks = Task.query.all()
+    else:
+        tasks = Task.query.filter_by(title=name_param)
+    
+    response = []
+
+    for task in tasks:
+        if task.completed_at is None:
+            task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False
+            }
+            response.append(task_dict)
+        else:
+            task_dict = {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "completed at": task.completed_at,
+                "is_complete": True}
```

Comment:
This could be compressed a bit you could just make a local variable set to be equal to `task.completed_at == None`